rcl/src/rcl/wait.c

```diff
@@ -79,6 +80,9 @@ __wait_set_clean_up(rcl_wait_set_t * wait_set, rcl_allocator_t allocator)
     assert(ret == RCL_RET_OK);  // Defensive, shouldn't fail with size 0.
   }
   if (wait_set->impl) {
+    if (wait_set->impl->rmw_waitset) {
+      assert(rmw_destroy_waitset(wait_set->impl->rmw_waitset) == RMW_RET_OK);
```

Comment:
I've decided the minimal viable solution here is to mirror what's in `publisher.c` and move the `rmw_destroy_waitset` call to `wait_set_fini`, which already has the right function signature to propagate an error code.